altapay.php

```diff
@@ -1865,7 +1873,7 @@ public function hookPayment($params)
 
         // Fetch payment methods
         $currency = $this->getCurrencyForCart($params['cart']);
-        $paymentMethods = Altapay_Models_Terminal::getActiveTerminalsForCurrency($currency->iso_code);
+        $paymentMethods = Altapay_Models_Terminal::getActiveTerminals((int) $this->context->shop->id);
```

Comment:
You are validating the the shop_id in the function, why is it being casted now?

Reply:
Done